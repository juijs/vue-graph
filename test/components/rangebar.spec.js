import { mount } from '@vue/test-utils'
import GraphRangeBar from '@/components/rangebar'

describe('rangebar.js', () => {
    it('snapshot', () => {
        const wrapper = mount(GraphRangeBar, {
            props: {
                width: 600,
                height: 400,
                axisMin: 0,
                axisMax: 50,
                barMargin: 30,
                labels: [ '1Q', '2Q', '3Q', '4Q' ],
                values: [
                    [ [3,25], [25,30], [30,43], [43,48] ],
                    [ [10,20], [20,25], [25,33], [33,45] ],
                    [ [5,35], [35,38], [38,40], [40,48] ]
                ]
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});