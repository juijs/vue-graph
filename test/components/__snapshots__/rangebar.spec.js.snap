// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rangebar.js snapshot 1`] = `
<div
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="400"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="600"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="300"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,50)"
        width="500"
        x="0"
        y="0"
      />
      <g
        class="grid-block"
        transform="translate(50,350)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            1Q
          </text>
        </g>
        <g
          transform="translate(125,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            2Q
          </text>
        </g>
        <g
          transform="translate(250,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            3Q
          </text>
        </g>
        <g
          transform="translate(375,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            4Q
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="500"
          y1="0"
          y2="0"
        />
        <g
          transform="translate(500,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
        </g>
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-range"
        transform="translate(50,50)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            50
          </text>
        </g>
        <g
          transform="translate(0,30)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            45
          </text>
        </g>
        <g
          transform="translate(0,60)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            40
          </text>
        </g>
        <g
          transform="translate(0,90)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            35
          </text>
        </g>
        <g
          transform="translate(0,120)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            30
          </text>
        </g>
        <g
          transform="translate(0,150)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            25
          </text>
        </g>
        <g
          transform="translate(0,180)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            20
          </text>
        </g>
        <g
          transform="translate(0,210)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            15
          </text>
        </g>
        <g
          transform="translate(0,240)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            10
          </text>
        </g>
        <g
          transform="translate(0,270)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            5
          </text>
        </g>
        <g
          transform="translate(0,300)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            0
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="0"
          y1="0"
          y2="300"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,50)"
      >
        <rect
          fill="transparent"
          height="300"
          stroke="transparent"
          width="500"
          x="0"
          y="0"
        />
      </g>
      <g
        class="brush-rangecolumn"
        transform="translate(50,50)"
      >
        <rect
          fill="#7977C2"
          height="132"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="30"
          y="150"
        />
        <rect
          fill="#7BBAE7"
          height="60"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="52"
          y="180"
        />
        <rect
          fill="#FFC000"
          height="180"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="74"
          y="90"
        />
        <rect
          fill="#7977C2"
          height="30"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="155"
          y="120"
        />
        <rect
          fill="#7BBAE7"
          height="30"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="177"
          y="150"
        />
        <rect
          fill="#FFC000"
          height="18"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="199"
          y="72"
        />
        <rect
          fill="#7977C2"
          height="78"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="280"
          y="42"
        />
        <rect
          fill="#7BBAE7"
          height="48"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="302"
          y="102"
        />
        <rect
          fill="#FFC000"
          height="12"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="324"
          y="60"
        />
        <rect
          fill="#7977C2"
          height="30"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="405"
          y="12"
        />
        <rect
          fill="#7BBAE7"
          height="72"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="427"
          y="30"
        />
        <rect
          fill="#FFC000"
          height="48"
          stroke="undefined"
          stroke-opacity="undefined"
          stroke-width="undefined"
          width="21"
          x="449"
          y="12"
        />
      </g>
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
